samples/dashboard/demos/climate/demo.js

```diff
@@ -749,9 +745,43 @@ async function convertTemperature(city) {
     });
 }
 
+async function setupCitiesData() {
+    const cities = citiesTable.modified;
+    const data = citiesData;
+
+    const rows = cities.getRows(
+        void 0,
+        void 0,
+        ['lat', 'lon', 'city', 'country', 'elevation']
+    );
+
+    for (const row of rows) {
+        const city = row[2];
+
+        if (typeof data[city] === 'undefined') {
+            data[city] = {
+                country: row[3],
+                elevation: row[4],
+                lat: row[0],
+                lon: row[1],
+                name: row[2],
+                store: await dataPool.getStore(city)
+            };
+
+            await convertTemperature(city);
+
+            if (citiesMap) {
+                citiesMap.setData(await buildCitiesMap());
+            }
+        }
+    }
+}
+
 async function main() {
     await setupDataPool();
     await setupDashboard();
+    await delay(750); // wait for animations to finish
```

Comment:
750 is an animation time from the chart? If yes, can we extract it from the chart options or set it as global?

Reply:
Not really. It is the timing of rendering plus animation and covers two charts - navigator and city. If we like to be exact then the follow up setupCitiesData needs to run after both chart complete callbacks. The best would be an animation finished callback though. I will create a promise wrap.